pkg/virt-controller/watch/pool/pool.go

```diff
@@ -661,60 +661,74 @@ func filterVMsBasedOnOrderedPolicies(vms []*virtv1.VirtualMachine, orderedPolici
 	return filteredVms, nil
 }
 
+func (c *Controller) proactiveScaleIn(pool *poolv1.VirtualMachinePool, vms []*virtv1.VirtualMachine, count int) error {
+	if isUnmanaged(pool) || isOpportunisticScaleInEnabled(pool) {
+		return nil
+	}
+
+	eligibleVMs := filterRunningVMs(vms)
+
+	// make sure we count already deleting VMs here during scale in.
+	count = count - (len(vms) - len(eligibleVMs))
+
+	return c.scaleIn(pool, eligibleVMs, count)
+}
+
 func (c *Controller) scaleIn(pool *poolv1.VirtualMachinePool, vms []*virtv1.VirtualMachine, count int) error {
+	if len(vms) == 0 || count == 0 {
+		return nil
+	} else if count > len(vms) {
+		count = len(vms)
+	}
 
 	poolKey, err := controller.KeyFunc(pool)
 	if err != nil {
 		return err
 	}
 
-	elgibleVMs := filterRunningVMs(vms)
+	eligibleVMs := filterRunningVMs(vms)
```

Comment:
Just a general tip: it would make it much easier to review if behavior changes and refactors reside on different commits.